src/toil/lib/accelerators.py

```diff
@@ -106,6 +107,30 @@ def count_nvidia_gpus() -> int:
     # TODO: Parse each gpu > product_name > text content and convert to some
     # kind of "model" that agrees with e.g. Kubernetes naming.
 
+@memoize
+def count_amd_gpus() -> int:
+    """
+    Return the number of amd GPUs seen by rocm-smi, or 0 if it is not working.
+    :return:
+    """
+    try:
+        # see if the amd-smi CLI tool is installed
+        # we believe this is the expected output for amd-smi, but we don't actually have and amd gpu to test against
+        out = subprocess.check_output((["amd-smi", "static"]))
+        gpu_count = len([line for line in out.decode("utf-8").split("\n") if line.startswith("gpu")])
+        return gpu_count
+    except (FileNotFoundError, subprocess.SubprocessError):
+        # if the amd-smi command fails, try rocm-smi
+        # if a different exception is raised, something other than the subprocess call is wrong
+        pass
+    try:
+        out = subprocess.check_output(["rocm-smi"])
+        gpu_count = len([line for line in out.decode("utf-8").split("\n") if len(line)> 0 and line[0] in string.digits])
```

Comment:
We should link to the example amd-smi and rocm-smi output that we were looking at, since it was hard to find and also necessary to anyone who wants to understand or adjust this parsing code.